backend/main.go

```diff
@@ -151,84 +149,86 @@ func isItemOrderedWithin30Days(orderDate string) (bool, error) {
 }
 
 // populates currente price and price drop in orderHistory slice
-func getPriceInfo(orderHistory *[]OrderInfo) error {
-	eg := new(errgroup.Group)
+func getPriceInfo(orderHistory *[]OrderInfo) {
+	var wg sync.WaitGroup
 	for _, item := range *orderHistory {
 		//we are starting a new goroutine
-		eg.Go(func() error {
-			return getPriceInfoForItem(&item)
-		})
+		wg.Add(1)
+		go func() {
+			//decrement the weight group counter once getPriceInfoForItem completes
+			defer wg.Done()
+			getPriceInfoForItem(&item)
+		}()
 	}
-	//the first goroutine that returns a non-nil error will output an error, otherwise we wait until all goroutines are finished.
-	if err := eg.Wait(); err != nil {
-		return errors.New("getPriceInfo: " + err.Error())
-	}
-
-	return nil
+	//wait for all threads to complete
+	wg.Wait()
 }
 
-func getPriceInfoForItem(item *OrderInfo) error {
+// if there is an error, we will just return early.
+// we want to keep getting prices. Errored items will get put into the unavailable category
+// possible error messages will be kept in every if statement for error handling
+func getPriceInfoForItem(item *OrderInfo) {
 	//webscraping portion!
 	//we first generate the URL we need to GET
 	itemUrl := getUrl(item.Name, item.Asin)
 
 	//now let's actually GET the webpage
 	resp, err := http.Get(itemUrl)
 	if err != nil {
-		return errors.New("getPriceInfoForItem: Cannot retrieve webpage for " + itemUrl + err.Error())
+		return
+		//errors.New("getPriceInfoForItem: Cannot retrieve webpage for " + itemUrl + err.Error())
 	}
-	if resp.Status != "200" {
-		return errors.New("Received HTTP status " + resp.Status + " when retrieving " + itemUrl)
+	if resp.Status != "200 OK" {
+		return
+		//errors.New("Received HTTP status " + resp.Status + " when retrieving " + itemUrl)
 	}
 
 	//let's look at the HTML body of the response
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return errors.New("getPriceInfoForItem: Cannot ready body of response from " + itemUrl + err.Error())
+		return
+		//errors.New("getPriceInfoForItem: Cannot ready body of response from " + itemUrl + err.Error())
 	}
 	respBodyString := string(body)
 	defer resp.Body.Close()
 
 	//We first check if we got served a captcha - the webpage will only have the following quote if the program was served a captcha
-	if strings.Index(respBodyString, `For information about migrating to our APIs refer to our Marketplace APIs at https://developer.amazonservices.com/ref=rm_c_sv, or our Product Advertising API at https://affiliate-program.amazon.com/gp/advertising/api/detail/main.html/ref=rm_c_ac for advertising use cases.`) != -1 {
-		return errors.New("getPriceInfoForItem: Got served a captcha by Amazon, likely for suspicious behavior regarding item: " + item.Name)
+	//This is exceedingly common by the way.
+	if strings.Index(respBodyString, CAPTCHA_INDICATOR) != -1 {
+		return

```

Comment:
time to figure out how to run the code on your local machine so you can hopefully avoid captcha checks :kek:

Reply:
also add a log